### PR TITLE
Debian packaging: the conclusion

### PR DESCRIPTION
**Pull request by aequitas**
This PR supersedes https://github.com/purpleidea/mgmt/pull/165.

Building on the work done by @dsx. I squashed it and made some small modifications. 

It can build a Debian package inside a `purpleidea/mgmt:development` Docker container using `make deb` and I have tested this deb on a Jessie system.

Some remarks:

- I'm all for ditching the Debian buildchain and opt for FPM as cross platform package solution.
- I replaced dsx with my name as maintainer, I don't know if you want your name in there or something else?
- I had to add some packages to `make-deps.sh`, is that ok or should I make it a flag (eg: via envvar BUILD_DEPS=1).

### CODE REVIEW

File: debian/copyright [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Format: http://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: mgmt
+Source: <https://github.com/purpleidea/mgmt>
+
+Files: *
+Copyright: Copyright (C) 2013-2018+ James Shubin and the project contributors
+License: AGPL-3.0
```
Comment (purpleidea) [inline debian/copyright]: We are actually GPLv3 now

File: misc/gen-deb-changelog-from-git.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	fi
+}
+
+if [ ! -d "debian" ]; then
+	echo "Directory ./debian not found"
+	exit 1
+fi
+
+tmpfile=$(mktemp)
+firstHash=$(git rev-list --max-parents=0 HEAD) # This should yeild the very first commit hash
```
Comment (purpleidea) [inline misc/gen-deb-changelog-from-git.sh]: s/yeild/yield/

**Comment by aequitas** [comment]
Both issues are adressed. I also added a copy of the COPYING file names LICENCE, this will cause github to pick it up and display the licence it as a feature of the repository: https://help.github.com/articles/adding-a-license-to-a-repository/

**Comment by purpleidea** [comment]
@aequitas Actually, while your intentions are good, please don't add the LICENSE file. It's actually a bug in the license script. Otherwise patch LGTM :)

**Comment by aequitas** [comment]
Ok, I removed the file. What is the bug btw?

**Comment by purpleidea** [comment]
LGTM-- they used to actually look at this file, I think a newer version of https://github.com/benbalter/licensee (which is the lib they use to do this) works again. Even though it would be nice to have them show this correctly, the trade off of monkeying unnecessarily with legal things makes lawyers nervous :(

**Comment by purpleidea** [comment]
Rebased to master and merged in https://github.com/purpleidea/mgmt/commit/94c8bc1de9aab32646f2190c4b9cf57e22d72efe

Thanks!!

**Comment by purpleidea** [comment]
@dsx Since @aequitas had continued off on your patch and ammended, it ended up with your name in the commit in master. I hope you don't mind!

Thanks to you both!

**Comment by dsx** [comment]
It's all good @purpleidea. Thanks to @aequitas for finishing it!

**Pull request closed without merging**